crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -295,6 +315,40 @@ async fn put_handler(
         })
 }
 
+fn override_helper(
+    req: Json<PutReq>,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    user: &User,
+) -> superposition::Result<Json<PutResp>> {
+    use contexts::dsl::contexts;
+    let new_ctx = create_ctx_from_put_req(req, conn, user)?;
+
+    let insert = diesel::insert_into(contexts).values(&new_ctx).execute(conn);
+
+    match insert {
+        Ok(_) => Ok(Json(get_put_resp(new_ctx))),
+        Err(DatabaseError(UniqueViolation, _)) => {
+            replace_override_of_existing_ctx(conn, new_ctx) // no need for .map(Json)
+        }
+        Err(e) => {
+            log::error!("failed to update context with db error: {:?}", e);
+            Err(db_error!(e))
+        }
+    }
+}
+
+#[put("/overrides")]
+async fn update_override_handler(
```

Comment:
Why do we need all this code? I thought update was straight forward and could be done with PUT context API

Reply:
The context PUT api will merge the overrides, it will not replace it. 
What we wanted was to replace the overrides through update api.